src/GenSymIO.chpl

```diff
@@ -452,18 +307,23 @@ module GenSymIO {
      * Reads all datasets from 1..n HDF5 files into an Arkouda symbol table. 
      */
     proc readAllHdfMsg(cmd: string, payload: string, st: borrowed SymTab): MsgTuple throws {
-        // reqMsg = "readAllHdf <ndsets> <nfiles> [<json_dsetname>] | [<json_filenames>]"
         var repMsg: string;
         // May need a more robust delimiter then " | "
-        var (strictFlag, ndsetsStr, nfilesStr, allowErrorsFlag, arraysStr) = payload.splitMsgToTuple(5);
+        var (strictFlag, ndsetsStr, nfilesStr, allowErrorsFlag, calcStringOffsetsFlag, arraysStr) = payload.splitMsgToTuple(6);
         var strictTypes: bool = true;
         if (strictFlag.toLower().strip() == "false") {
           strictTypes = false;
         }
 
         var allowErrors: bool = "true" == allowErrorsFlag.toLower(); // default is false
         if allowErrors {
-            gsLogger.warn(getModuleName(), getRoutineName(), getLineNumber(), "Allowing file read errors");            
+            gsLogger.warn(getModuleName(), getRoutineName(), getLineNumber(), "Allowing file read errors");
+        }
+
+        var calcStringOffsets: bool = "true" == calcStringOffsetsFlag.toLower(); // default is false
```

Comment:
This is just personal preference, but it took my brain a second to catch up here. At first glance I read this as setting `calcStringOffsets` to true. I think something like this is slightly more readable
```C
var calcStringOffsets: bool = (calcStringsOffsetsFlag.toLower() == "true");  // default is false
```